commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -122,101 +120,81 @@ public static boolean executeInDir(String command, File commandOutputFile, Strin
     }
 
     /**
-     * Executes command in the directory specified, uses current directory if not
-     * specified If file provided directs output to file (creates if does not exist)
-     * else Asynchronously copy command output in
-     * {@link CommandResult#setCommandOutput(String)} waits for the process to
-     * complete
-     * if standard input is specified, it is passed as input to the command.
-     *
+     * Uses apache commons exec to execute passed command
+     * 
      * @param command
-     * @param file
+     * @param outputFile output stream for process {@link #getOutputStream(File)}
      * @param dir
      * @param stdInput
      * @return {@link CommandResult}
-     * @throws IOException
-     * @throws InterruptedException
      * @throws HyscaleException
+     * @throws IOException
      */
-    private static CommandResult _execute(String command, File file, String dir, String stdInput)
-            throws IOException, InterruptedException, HyscaleException {
+    private static CommandResult _execute(String command, File outputFile, String dir, String stdInput)
+            throws HyscaleException, IOException {
+        
         CommandResult cmdResult = new CommandResult();
-        int exitCode = 1;
-        cmdResult.setExitCode(exitCode);
         if (StringUtils.isBlank(command)) {
             return cmdResult;
         }
-        ProcessBuilder processBuilder = new ProcessBuilder();
         if (StringUtils.isBlank(dir) || dir.equals(".")) {
             dir = SetupConfig.CURRENT_WORKING_DIR;
         }
-        logger.debug("Executing command {} in dir {}", command, dir);
-        processBuilder.command(command.split(" "));
-        processBuilder.redirectErrorStream(true);
-        if (file != null) {
-            HyscaleFilesUtil.createEmptyFile(file);
-            processBuilder.redirectOutput(file);
-        }
-        Process process = processBuilder.start();
-        handleStandardInput(process, command, stdInput);
+        CommandLine commandToExecute = CommandLine.parse(command);
 
-        boolean readOutput = false;
-        StringWriter strWriter = new StringWriter();
-        if (file == null) {
-            readOutput = copyOutput(process, strWriter);
-        }
+        Executor executor = new DefaultExecutor();
+
```

Comment:
Why not a daemon executor ?

Reply:
WE can take it up as additional functionality later on